pyTigerGraph/gds/featurizer.py

```diff
@@ -0,0 +1,189 @@
+        To add a specific algorithm, it should be added both to the gsql folder and class variable dictionary. 
+        Args:
+            conn (TigerGraphConnection): Connection to the TigerGraph database.
+            result_attr (str): An attribute name that needs to be added to the vertex/edge
+            local_gsql_path (str): A path to where the algorithm. is located
+        """
+
+        self.conn = conn
+        self.name_of_query = name_of_query
+        self.result_attr = result_attr
+        self.local_gsql_path = local_gsql_path
+        self.queryResult_type_dict = {"tg_pagerank":"Float","tg_fastRP":"List<Double>","tg_label_prop":"INT","tg_louvain":"INT"}#List of graph algorithms
+        self.params_dict = {}#input parameter for the desired algorithm to be run
+
+    def _is_query_installed(self, query_name: str) -> bool:
+        #If the query id already installed return true
+        self.name_of_query = query_name
+        resp = "GET /query/{}/{}".format(self.conn.graphname, query_name)
+        queries = self.conn.getInstalledQueries()
+        return resp in queries
+
+    def _install_query_file(self, file_path: str, replace: dict = None):
+        #Read the first line of the query file to get the query name, e.g, CREATE QUERY query_name ...
+        self.local_gsql_path = file_path
+        with open(file_path) as infile:
+            firstLine = infile.readline()
+        try:
+            name_of_query = firstLine.split("QUERY")[1].strip().split("(")[0]
+        except:
+            raise ValueError("Cannot parse the query file. The query file should start with CREAT QUERY query_name ...")
+        # If a suffix is to be added to query name
+        if replace and ("{QUERYSUFFIX}" in replace):
+            name_of_query = name_of_query.replace("{QUERYSUFFIX}",replace["{QUERYSUFFIX}"])
+        # If query is already installed, skip.
+        if self._is_query_installed(name_of_query.strip()):
+            return name_of_query
+        # Otherwise, install query from file.
+        with open(file_path) as infile:
+            query = infile.read()
+        # Replace placeholders with actual content if given
+        if replace:
+            for placeholder in replace:
+                query = query.replace(placeholder, replace[placeholder])
+        # TODO: Check if Distributed query is needed.
+        query = ("USE GRAPH {}\n".format(self.conn.graphname) + query + "\ninstall Query {}\n".format(name_of_query))
+        print("Installing and optimizing the queries, it might take a minute")
+        resp = self.conn.gsql(query)
+        status = resp.splitlines()[-1]
+        if "Failed" in status:
+            raise ConnectionError(status)
+        return name_of_query 
+
+    def installAlgorithm(self,query_name:str,schema_type:str="VERTEX",attr_name:str=None,schema_name:str=None):
+        query_path = pjoin(os.path.dirname(os.path.abspath(__file__)), "gsql", "featurizer", query_name+'.gsql')
```

Comment:
Might want to check if the input query_name is in the allowlist of query names

Reply:
added the check